### PR TITLE
GPT2Block as a separate construction unit

### PR DESCRIPTION
**Pull request by glkarpov**

### CODE REVIEW

File: wrappers/python/nntile/model/gpt2_block.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+#                2023-present Artificial Intelligence Research Institute
+#                              (AIRI), Russia. All rights reserved.
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/nntile/model/gpt2_block.py
+# GPT2Block submodule of NNTile Python package
+#
+# @version 1.0.0
```
Comment (Muxas) [inline wrappers/python/nntile/model/gpt2_block.py:12]: version 1.1.0

File: wrappers/python/nntile/model/gpt2_block.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            resid_pdrop=0.0,
+            embd_pdrop=0.0,
+            attn_pdrop=0.0,
+            layer_norm_epsilon=self.config.layer_norm_epsilon,
+            scale_attn_weights=True,
+            use_cache=False,
+            add_cross_attention=False,
+        )
+        gpt2_block_torch = GPT2Block_torch(config_torch)
+        gpt2_block_torch.ln_1 = self.layers[0].to_torch()
```
Comment (Muxas) [inline wrappers/python/nntile/model/gpt2_block.py:118]: add a name to `self.layers[0]` as it is more readable to access elements by name

File: wrappers/python/nntile/model/gpt2_block.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        gpt2_block_torch.attn = self.layers[1].to_torch()
```
Comment (Muxas) [inline wrappers/python/nntile/model/gpt2_block.py:119]: add a name to `self.layers[1]`

File: wrappers/python/nntile/model/gpt2_block.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        p_attn_n = self.layers[3]
```
Comment (Muxas) [inline wrappers/python/nntile/model/gpt2_block.py:120]: add a name to `self.layers[3]`

File: wrappers/python/nntile/model/gpt2_block.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            resid_pdrop=0.0,
+            embd_pdrop=0.0,
+            attn_pdrop=0.0,
+            layer_norm_epsilon=self.config.layer_norm_epsilon,
+            scale_attn_weights=True,
+            use_cache=False,
+            add_cross_attention=False,
+        )
+        gpt2_block_torch = GPT2Block_torch(config_torch)
+        gpt2_block_torch.ln_1 = self.layers[0].to_torch_with_grads()
```
Comment (Muxas) [inline wrappers/python/nntile/model/gpt2_block.py:141]: named `self.layers[i]`

File: wrappers/python/tests/model/test_gpt2_block.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+#
+# NNTile is software framework for fast training of big neural networks on
+# distributed-memory heterogeneous systems based on StarPU runtime system.
+#
+# @file wrappers/python/tests/layer/test_gpt2_block.py
+# Test for nntile.model.gpt2_block
+# Each test is generated in float precision by PyTorch, then it is downcasted
+# into NNTile type. So, implementation of double precision is NOT checked.
+#
+# @version 1.0.0
```
Comment (Muxas) [inline wrappers/python/tests/model/test_gpt2_block.py:14]: version 1.1.0

**Review by Muxas — CHANGES_REQUESTED** [review]
Access layers by names

File: wrappers/python/nntile/model/gpt2_block.py (hunk 1 above)
Comment (daskol) [reply wrappers/python/nntile/model/gpt2_block.py:12]: What about `@version 1.0.0` in all source files of release tag `v1.1.0`?

Comment (daskol) [reply wrappers/python/nntile/model/gpt2_block.py:12]: That's insane. My browser crashed, lol.

image

**Review by Muxas — APPROVED** [review]

**Pull request merged**